nvflare/app_common/statistics/hierarchical_numeric_stats.py

```diff
@@ -0,0 +1,477 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import copy
+from math import sqrt
+from typing import Dict, List, TypeVar
+
+from nvflare.app_common.abstract.statistics_spec import Bin, BinRange, DataType, Feature, Histogram, HistogramType
+from nvflare.app_common.app_constant import StatisticsConstants as StC
+
+T = TypeVar("T")
+
+
+def get_initial_structure(client_metrics: dict, metric_task: str, ordered_metrics: dict) -> dict:
+    """Calculate initial output structure that is common at all the hierarchical levels."""
+    stats = {}
+    for metric in ordered_metrics:
+        stats[metric] = {}
+        for stat in client_metrics:
+            for device in client_metrics[stat]:
```

Comment:
It's actually just the variable name and all the output is based on the hierarchical configuration. But let me just recheck this and change the variable name as well to avoid the confusion, thanks.

Reply:
Done, changed variable name, thanks.